### ISSUE TITLE
[feature]: change `openchannel` to make private channels by default

### ISSUE THREAD

**Issue by AndySchroder** [lightningnetwork/lnd#9157]
I believe that the `openchannel` command should make private channels by default.

With the addition of blinded paths, I think there are a lot of nodes that won't want to announce their channel peers and one should need to make an explicit choice by default to announce channels publicly.

Of course, there should be an option in `lnd.conf` to change this default, but I believe that default value should be set to private by default. The autopilot feature already has this (https://github.com/lightningnetwork/lnd/blob/9f0cc159ea04edda9f101a35693b8432a31854e1/sample-lnd.conf#L882).

Also, if we get some protocol changes and https://github.com/lightningnetwork/lnd/issues/7902 gets fixed, you could change your mind later if you want to make it public, giving a lot more flexibility to evaluate if the channel partner is a good one before announcing it (it is never going to be possible to make a protocol change to turn a public channel private). Do dynamic commitments help with making https://github.com/lightningnetwork/lnd/issues/7902 possible?

**Comment by saubyk** [comment]
Shouldn't this be a prerogative of the applications instead?
LND has many types of users from routing nodes to wallets to payment processors to exchanges. This might be preference for one cohort (wallets?), but the rest may not want this as default.

**Comment by AndySchroder** [comment]
Yes, that's why I proposed the default could be something that can be set in the `lnd.conf` file. It seems to me like LND should default to be safe for even the most unfamiliar and new users and keeping channels private unless explicitly opting in to make them public seems to be the safest and most conservative option in my opinion. This is surely up for debate, that's why I wanted to open this issue to see what other people thought.

Even if you don't agree that the default default should be private, would you agree that each operator should be able to set their personal default to private in `lnd.conf`?
